File: example-ruby/assignment1.rb

class Grader
   def initialize(name, assignment, grade)
    @read_grade = ""
    @grades = []
    # @letter_grade =""
    @student = name
    @student_assignment = assignment
    @student_grade = grade
   end

   def display_student(grade)
    @student_grade = grade
     if @student_grade >= 90.00
      @read_grade =  @grades.push('A')      
     elsif @student_grade >= 80.00
        @read_grade =  @grades.push('B')
     elsif  @student_grade >= 80.00
        @read_grade =  @grades.push('C')
     elsif @student_grade >= 60.00
        @read_grade =  @grades.push('D')
     elsif @student_grade < 60.00
         @read_grade = @grades.push('F')

     end
   end
    def display_details()
      puts "Student's Name:  #@student"
      puts "student's assignment: #@student_assignment"
      puts "student's grade: #@read_grade" 
    #   puts "student's grades: #@letter_grade" 
   end
 end
puts "Please enter student's name \n"
@name = gets
puts "Please enter student's assignment \n"
@assignment = gets
puts "Please enter student's grade \n"
@grade = gets

def main 
showStudent = Grader.new(@name, @assignment, @grade)
showStudent.display_student(@grade.to_f.round(2))
showStudent.display_details()
end

class Object
  def is_number?
    to_f.to_s == to_s || to_i.to_s == to_s
  end
  def numeric?    
    Float(self) != nil rescue false
  end
end


if @grade.numeric?
    puts "grade is a number."
    main
else 
    puts "Not a number"
    puts "Please enter a number."
 end

File: example-python/assignment1.py
# 100-90=A, 80-89=B, 79-70=C, 69-60=D, <60=F
class Grader:
    def __init__(self, student, assignment, grade):
        self.grades = []
        self.student = student
        self.assignment = assignment
        self.grade = grade
    
    def displayStudent(self):
        if self.grade >= 90.00:
         self.grades.append('A')
        elif self.grade >= 80.00:
          self.grades.append('B')
        elif self.grade >= 70.00:
          self.grades.append('C')
        elif self.grade >= 60.00:
          self.grades.append('D')
        elif self.grade < 60.00:
          self.grades.append('F')
        print("Student's Name: ", self.student,  "\nAssignment: ",
              self.assignment,  "\nGrade: ", str(self.grades[0]))
     
# commandline requests

requestName = input("Please enter student's name: ")
requestAssignment = input("Please enter student's assignment: ")
requestGrade = input("Please enter grade:")

def getMain():
  showStudent = Grader(requestName, requestAssignment, requestGrade)
#   showStudent = Grader( requestName, requestAssignment, requestGrade)
  showStudent.displayStudent()

#get the true on value if not right
res = True
while res:
  try : 
    requestGrade = float(requestGrade)
    requestGrade = round(requestGrade,2)
    # res = True
    getMain()
    res = False
  except ValueError:
    print("Not a float")
    print("Please enter a number.")
    requestGrade = input("Please enter grade:")
    # res = False

"""

Local Machine:

aimeelynnramirez@Aimees-MBP example-python % python assignment1.py
Please enter student's name: aimee
Please enter student's assignment: test
Please enter grade:94
Student's Name:  aimee 
Assignment:  test 
Grade:  A
aimeelynnramirez@Aimees-MBP example-python % python assignment1.py
Please enter student's name: aimee
Please enter student's assignment: test
Please enter grade:54
Student's Name:  aimee 
Assignment:  test 
Grade:  F
aimeelynnramirez@Aimees-MBP example-python % python assignment1.py
Please enter student's name: aimee
Please enter student's assignment: test
Please enter grade:89.9
Student's Name:  aimee 
Assignment:  test 
Grade:  B
aimeelynnramirez@Aimees-MBP example-python % python assignment1.py
Please enter student's name: aimee
Please enter student's assignment: test 
Please enter grade: 60.01
Student's Name:  aimee 
Assignment:  test 
Grade:  D
aimeelynnramirez@Aimees-MBP example-python % python assignment1.py
Please enter student's name: aimee
Please enter student's assignment: test
Please enter grade:102.1
Student's Name:  aimee 
Assignment:  test 
Grade:  A

SSH Machine:
vagrant@ubuntu-focal:/var/www/html/wdd353-grader/example-python$ python3 assignment1.py 
Please enter student's name: aimee
Please enter student's assignment: test
Please enter grade:94
Student's Name:  aimee 
Assignment:  test 
Grade:  A
vagrant@ubuntu-focal:/var/www/html/wdd353-grader/example-python$ python3 assignment1.py 
Please enter student's name: 54
Please enter student's assignment: 54
Please enter grade:54
Student's Name:  54 
Assignment:  54 
Grade:  F
vagrant@ubuntu-focal:/var/www/html/wdd353-grader/example-python$ python3 assignment1.py 
Please enter student's name: aimee
Please enter student's assignment: test
Please enter grade:94
Student's Name:  aimee 
Assignment:  test 
Grade:  A
vagrant@ubuntu-focal:/var/www/html/wdd353-grader/example-python$ python3 assignment1.py 
Please enter student's name: aimee
Please enter student's assignment: test
Please enter grade:54
Student's Name:  aimee 
Assignment:  test 
Grade:  F
vagrant@ubuntu-focal:/var/www/html/wdd353-grader/example-python$ python3 assignment1.py 
Please enter student's name: aimee
Please enter student's assignment: test
Please enter grade:89.9
Student's Name:  aimee 
Assignment:  test 
Grade:  B
vagrant@ubuntu-focal:/var/www/html/wdd353-grader/example-python$ python3 assignment1.py 
Please enter student's name: test
Please enter student's assignment: test1
Please enter grade:60.01
Student's Name:  test 
Assignment:  test1 
Grade:  D
vagrant@ubuntu-focal:/var/www/html/wdd353-grader/example-python$ python3 assignment1.py 
Please enter student's name: aimee
Please enter student's assignment: test
Please enter grade:102.01
Student's Name:  aimee 
Assignment:  test 
Grade:  A

"""